src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -59,6 +79,17 @@ struct uct_dc_mlx5_ep {
     uint16_t                      atomic_mr_offset;
     uct_rc_fc_t                   fc;
     uct_ib_mlx5_base_av_t         av;
+
+#if UCS_ENABLE_ASSERT
+    struct {
+        const char                *file; /* Filename where DCI was allocated
+                                          * last time */
+        unsigned                  line; /* Line number where DCI was allocated
+                                         * last time */
+        int                       dci_waitq; /* Indicates if DCI was allocated
+                                              * from waiting DCI allocation */
+    } dci_alloc;
+#endif
```

Comment:
imo, it is unnecessarily complicates the code

Reply:
> maybe just use `const char *name` as an extra arg for debug purposes? for example like we do for ucs_alloc

not sure that I understand the idea.
where should it be saved to then print it when DCI leak detected?

Reply:
> not sure that I understand the idea.
> where should it be saved to then print it when DCI leak detected?

instead of `_ep->dci_alloc`